app/services/HolderOfAuthorisationCode.scala

```diff
@@ -18,9 +18,24 @@ package services
 
 import utils.FileReader
 
-case class HolderOfAuthorisationCode(value: String)
+import scala.util.matching.Regex
+
+case class HolderOfAuthorisationCode(code: String, description: String) {
+
+  def asString = s"${description} (${code})"
+}
 
 object HolderOfAuthorisationCode {
+
+  private val regex: Regex = """^"?(\w+)"?,"?([^"\n]+)"?$""".r
```

Comment:
Yes, that is correct.  The regex is based on similar list-from-csv-files classes (e.g. CustomsOffices) but I enhanced it so that it can cope with the "code" being quoted or not.
To be honest, I was surprised to see how we handle existing csv files and that we are not using some sort of library.
Alternatively, I could have used json I suppose.  There was no requirement to use csv.  The "source" is a Google doc so I had to create the csv file anyway.

Reply:
We have library `scala-csv`. Make everything really easy, but I added this library at the end, when we had a lot of difference codes, so some of the code lists are still handled by `FileReader`

CSV library is really good, the only cons is that they return `List[List[String]]`, so list of rows and every row list of values, so you need to use pattern matching to retrieve the values, but still I think it's good

Reply:
I didn't spot that library!   Doh!. 

I will change my PR to use it then.  As there isn't much else on shall I change the other regex-based code-list readers as well?

Reply:
Yeah, if you want :)